### PR TITLE
GRAM: fixed parsing of lifetime bound on Fn bound when it returns reference

### PR DESCRIPTION
**Pull request by rrevenantt**
In this case:
```rust
fn test<'b, F, Z: 'b>() where F: Fn() -> &'b Z + 'static {}
```
`'static` was parsed incorrectly as part of return type, while it should be part of `Fn` bound

### CODE REVIEW

File: src/main/grammars/RustParser.bnf [hunk 1]
```diff
@@ -857,7 +857,9 @@ ArrayType ::= '[' TypeReference [';' AnyExpr] ']' {
   elementTypeFactory = "org.rust.lang.core.stubs.StubImplementationsKt.factory"
 }
 
-RefLikeType ::= ( '&' Lifetime? mut? | '*' [ const | mut ] ) TypeReference {
+private RefLikeInner ::= ( '&' Lifetime? mut? | '*' [ const | mut ] )
```
Comment (vlad20012) [inline src/main/grammars/RustParser.bnf]: Let's rename to `RefLikeStart` like `PathStart`

**Comment by Undin** [comment]
@rrevenantt in #5065 we've introduce `org.rust.lang.core.parser.RustParserDefinition#PARSER_VERSION` that's supposed to be increased after each change in grammar to drop the corresponding indices and avoid possible bugs because of it
Could you please bump it?

**Comment by Undin** [comment]
bors r=vlad20012

**Comment by Undin** [comment]
bors retry

**Pull request merged**